appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -66,7 +85,12 @@ internal class ExperienceRenderer(
 
         if (!canShow) return false
 
-        val stateMachine = renderContextManager.getOrCreateStateMachines(renderContext)
+        val stateMachine = stateMachines.getOwner(renderContext)?.stateMachine
+
+        if (stateMachine == null) {
+            analyticsTracker.trackExperienceError(experience, "no render context $renderContext")
```

Comment:
this is the new spot where we track the exact error in question, and avoid propagating other show failures that are already handled elsewhere

Reply:
ok so instead of creating we just get instead, if null then we know the render context is not here. ok